src/mlpack/methods/ann/layer/conv_layer.hpp

```diff
@@ -126,7 +132,11 @@ class ConvLayer
         BackwardConvolutionRule::Convolution(gy.slice(inMap), rotatedFilter,
             output);
 
-        g.slice(outMap) += output;
+        if (wPad != 0 || hPad != 0)
```

Comment:
The gradient isn't set if ``(wPad != 0 || hPad != 0)`` isn't true. In case ``(wPad != 0 || hPad != 0)``is false the gradient is the output value.